server/api/users/users.controller.ts

```diff
@@ -29,9 +29,32 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 };
 
 export const getUsers: RequestHandler<UserProfile[]> = async (req, res) => {
-  const users = await User.findAll({ attributes: User.allowedFields });
+  const page = req.query.page ? parseInt(req.query.page as string, 10) : 1;
+  const limit = 20;
+  const offset = (page - 1) * limit;
 
-  res.json(users);
+  if (!Number.isInteger(page) || page < 1) {
+    return res.status(400).json({ error: 'Invalid page number' });
+  }
```

Comment:
You've already made sure `page` will be an integer, therefore you wouldn't need `!Number.isInteger(page)`.

```suggestion
  if (page < 1) {
    return res.status(400).json({ error: 'Invalid page number' });
  }
```

Additionally, if we default to `1` above when invalid input is provided, we can eliminate this block entirely.
```suggestion
```

Reply:
It may not be an integer - `parseInt` can return `NaN` if a non-integer-like value is passed.  e.g. running the following in your browser

```js
Number.isInteger(parseInt("asdf"))
```

Reply:
Ahh, good catch. Wouldn't have to worry about it though if we defaulted to 1 when not given a valid integer. 😜